processing/src/main/java/io/druid/query/extraction/MatchingDimExtractionFn.java

```diff
@@ -77,6 +77,12 @@ public boolean preservesOrdering()
   }
 
   @Override
+  public ExtractionType getExtractionType()
+  {
+    return ExtractionType.ONE_TO_ONE;
```

Comment:
I'll change it to replicate current behavior (`MANY_TO_ONE`) and let any optimizations be handled in a future PR.